applications/integrationtests/mechanics/QuasistaticProblem1D.cpp

```diff
@@ -0,0 +1,224 @@
+
+#include "mechanics/constitutive/LinearElastic.h"
+#include "mechanics/constitutive/LocalIsotropicDamage.h"
+#include "mechanics/constitutive/damageLaws/DamageLawExponential.h"
+#include "mechanics/constitutive/ModifiedMisesStrainNorm.h"
+
+#include "visualize/Visualizer.h"
+#include "visualize/AverageHandler.h"
+#include "visualize/AverageGeometries.h"
+
+using namespace NuTo;
+
+
+class LocalDamageTruss
+{
+public:
+    LocalDamageTruss(int numElements)
+        : mMesh(UnitMeshFem::CreateLines(numElements))
+        , mDof("Dispacement", 1)
+        , mLaw(ConstitutiveLaw())
+        , mMomentumBalance(mDof, mLaw)
+        , mEquations(&mMesh)
+        , mProblem(mEquations, mDof)
+        , mIntegrationType(2, eIntegrationMethod::GAUSS)
+    {
+        AddDofInterpolation(&mMesh, mDof);
+        auto constraints = DefineConstraints(mMesh, mDof);
+        auto dofInfo = DofNumbering::Build(mMesh.NodesTotal(mDof), mDof, constraints);
+
+        mEquations.SetDofInfo(dofInfo);
+        mNumIndependentDofs = dofInfo.numIndependentDofs[mDof];
+        mProblem.SetConstraints(constraints, mNumIndependentDofs);
+        mProblem.mTolerance = 1.e-12;
+
+        mCellGroup = mCells.AddCells(mMesh.ElementsTotal(), mIntegrationType);
+        mLaw.mEvolution.ResizeHistoryData(mCellGroup.Size(), mIntegrationType.GetNumIntegrationPoints());
+
+
+        auto Gradient = Bind(mMomentumBalance, &Integrands::MomentumBalance<1>::Gradient);
+        auto Hessian0 = Bind(mMomentumBalance, &Integrands::MomentumBalance<1>::Hessian0);
+        CellInterface::VoidFunction UpdateHistory = [&](const CellData& cellData, const CellIpData& cellIpData) {
+            EngineeringStrain<1> strain = cellIpData.GetBMatrixStrain(mDof) * cellData.GetNodeValues(mDof);
+            mLaw.Update(strain, 0, cellData.GetCellId(), cellIpData.GetIpId());
+        };
+
+        mEquations.AddGradientFunction(mCellGroup, Gradient);
+        mEquations.AddHessian0Function(mCellGroup, Hessian0);
+        mEquations.AddUpdateFunction(mCellGroup, UpdateHistory);
+    }
+
+    void SetImperfection(double kappaImperfection)
+    {
+        const int imperfectionCell = mMesh.Elements.Size() / 2;
+        const int imperfectionIp = mLaw.mEvolution.Ip(imperfectionCell, 0);
+        mLaw.mEvolution.mKappas[imperfectionIp] = kappaImperfection;
+    }
+
+    void Solve(double tEnd)
+    {
+        Eigen::SimplicialLDLT<Eigen::SparseMatrix<double>> pureSolver;
+        auto solver = NewtonRaphson::CreateWrappedSolver(pureSolver);
+        Eigen::VectorXd x = Eigen::VectorXd::Zero(mNumIndependentDofs);
+
+        double globalTime = 0;
+        double timeStep = 0.01;
+        double maxTimeStep = 0.1;
+
+        int iStep = 0;
+        while (globalTime < tEnd && timeStep > 1.e-6)
+        {
+            std::cout << "Step " << iStep << " at t = " << globalTime << std::endl;
+            try
+            {
+                auto trialSystem = mProblem.TrialSystem(x, globalTime, timeStep);
```

Comment:
I thought about that and liked it until I wanted to implement it. If we chose to do `DoStep`, we have a lot of options to pass that are more suitable as template parameters. E.g. the solver or the solution scheme. So the default case would be to use a Newton-Raphson algorithm in this `DoStep`. Now it is not trivial do specify the solver. Another option is a different iteration scheme. What if I want to use a fix-point iteration or L-BGFS instead of Newton-Raphson.

Maybe providing `DoStep` as an additional method (and keeping the others as well) could be an option.